{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyNrdnQ/c0qzKzmCfPN8FxRm"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["fisrt we get the required data"],"metadata":{"id":"qbbkFs3veVZr"}},{"cell_type":"code","execution_count":23,"metadata":{"id":"VhjzJzB8ePGf","executionInfo":{"status":"ok","timestamp":1673528596876,"user_tz":-120,"elapsed":21120,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"18c6adc3-dbeb-4a64-e708-a09f8e8cebd1"},"outputs":[{"output_type":"stream","name":"stdout","text":["--2023-01-12 13:02:55--  https://ftp.ensembl.org/pub/release-108/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n","Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.139\n","Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.139|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 11389810 (11M) [application/x-gzip]\n","Saving to: ‘Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz’\n","\n","Homo_sapiens.GRCh38 100%[===================>]  10.86M   692KB/s    in 16s     \n","\n","2023-01-12 13:03:12 (683 KB/s) - ‘Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz’ saved [11389810/11389810]\n","\n","--2023-01-12 13:03:13--  https://github.com/PU-SDS/ngs-book-dataset/blob/master/chapter8/datasets.zip?raw=true\n","Resolving github.com (github.com)... 140.82.114.3\n","Connecting to github.com (github.com)|140.82.114.3|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://github.com/PU-SDS/ngs-book-dataset/raw/master/chapter8/datasets.zip [following]\n","--2023-01-12 13:03:13--  https://github.com/PU-SDS/ngs-book-dataset/raw/master/chapter8/datasets.zip\n","Reusing existing connection to github.com:443.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://raw.githubusercontent.com/PU-SDS/ngs-book-dataset/master/chapter8/datasets.zip [following]\n","--2023-01-12 13:03:13--  https://raw.githubusercontent.com/PU-SDS/ngs-book-dataset/master/chapter8/datasets.zip\n","Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.110.133, 185.199.108.133, ...\n","Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 38347854 (37M) [application/zip]\n","Saving to: ‘datasets.zip?raw=true’\n","\n","datasets.zip?raw=tr 100%[===================>]  36.57M   163MB/s    in 0.2s    \n","\n","2023-01-12 13:03:14 (163 MB/s) - ‘datasets.zip?raw=true’ saved [38347854/38347854]\n","\n","Archive:  data.zip\n","   creating: datasets/\n","  inflating: datasets/adaptor.fa     \n","   creating: __MACOSX/\n","   creating: __MACOSX/datasets/\n","  inflating: __MACOSX/datasets/._adaptor.fa  \n","  inflating: datasets/brain1_R1.fastq  \n","  inflating: datasets/brain1_R2.fastq  \n","  inflating: datasets/brain2_R1.fastq  \n","  inflating: datasets/brain2_R2.fastq  \n","  inflating: datasets/uhr1_R1.fastq  \n","  inflating: datasets/uhr1_R2.fastq  \n","  inflating: datasets/uhr2_R1.fastq  \n","  inflating: datasets/uhr2_R2.fastq  \n"]}],"source":["# get the reference genome\n","!wget https://ftp.ensembl.org/pub/release-108/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n","!gzip -d Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n","!mv Homo_sapiens.GRCh38.dna.chromosome.22.fa chr22.fa\n","!wget https://github.com/PU-SDS/ngs-book-dataset/blob/master/chapter8/datasets.zip?raw=true\n","!mv datasets.zip?raw=true data.zip\n","!unzip data.zip"]},{"cell_type":"markdown","source":["install required tools using conda"],"metadata":{"id":"D_cmqZ_tp-yB"}},{"cell_type":"code","source":["#install conda\n","#!wget -c https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","#!chmod +x Miniconda3-latest-Linux-x86_64.sh\n","#!bash ./Miniconda3-latest-Linux-x86_64.sh -b -f -p /usr/local\n","\n","#install fastqc version 0.11.5\n","#!printf \"y\" | conda install -c bioconda fastqc=0.11.5\n","\n","#install trimmomatic version 0.36\n","#!printf \"y\" | conda install -c bioconda trimmomatic=0.36\n","\n","#install STAR v 2.4.2\n","#!printf \"y\" | conda install -c bioconda star=2.4.2\n","\n","#install samtools v1.3\n","!printf \"y\" | conda install -c bioconda samtools=1.3\n","\n","#install cufflinks v2.2.1\n","#!printf \"y\" | conda install -c bioconda cufflinks=2.2.1\n","\n","#install HTSeq v0.6.1\n","!printf \"y\" | conda install -c bioconda htseq"],"metadata":{"id":"VHThmAJTfHTR","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673528211601,"user_tz":-120,"elapsed":81701,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"daa4372b-b60e-442f-ab0c-6b369a8bbcaf"},"execution_count":12,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - samtools=1.3\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    c-ares-1.18.1              |       h7f8727e_0         114 KB\n","    curl-7.86.0                |       h5eee18b_0          87 KB\n","    krb5-1.19.2                |       hac12032_0         1.2 MB\n","    libcurl-7.86.0             |       h91b91d3_0         368 KB\n","    libedit-3.1.20221030       |       h5eee18b_0         181 KB\n","    libev-4.33                 |       h7f8727e_1         111 KB\n","    libnghttp2-1.46.0          |       hce63b2e_0         680 KB\n","    libssh2-1.10.0             |       h8f2d780_0         274 KB\n","    samtools-1.3.1             |       h1b8c3c0_8         470 KB  bioconda\n","    ------------------------------------------------------------\n","                                           Total:         3.5 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  c-ares             pkgs/main/linux-64::c-ares-1.18.1-h7f8727e_0 \n","  curl               pkgs/main/linux-64::curl-7.86.0-h5eee18b_0 \n","  krb5               pkgs/main/linux-64::krb5-1.19.2-hac12032_0 \n","  libcurl            pkgs/main/linux-64::libcurl-7.86.0-h91b91d3_0 \n","  libedit            pkgs/main/linux-64::libedit-3.1.20221030-h5eee18b_0 \n","  libev              pkgs/main/linux-64::libev-4.33-h7f8727e_1 \n","  libnghttp2         pkgs/main/linux-64::libnghttp2-1.46.0-hce63b2e_0 \n","  libssh2            pkgs/main/linux-64::libssh2-1.10.0-h8f2d780_0 \n","  samtools           bioconda/linux-64::samtools-1.3.1-h1b8c3c0_8 \n","\n","\n","Proceed ([y]/n)? \n","\n","Downloading and Extracting Packages\n","libcurl-7.86.0       | 368 KB    | :   0% 0/1 [00:00<?, ?it/s]\n","libnghttp2-1.46.0    | 680 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n","\n","samtools-1.3.1       | 470 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n","\n","\n","libedit-3.1.20221030 | 181 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","c-ares-1.18.1        | 114 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","libev-4.33           | 111 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","curl-7.86.0          | 87 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","libssh2-1.10.0       | 274 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","\n","krb5-1.19.2          | 1.2 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","libcurl-7.86.0       | 368 KB    | :  30% 0.30471979849723146/1 [00:00<00:00,  2.93it/s]\n","\n","\n","libedit-3.1.20221030 | 181 KB    | :  62% 0.6180509150481774/1 [00:00<00:00,  6.04it/s]\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","c-ares-1.18.1        | 114 KB    | :  56% 0.5627194903102273/1 [00:00<00:00,  5.61it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","libev-4.33           | 111 KB    | :  14% 0.14370543193200655/1 [00:00<00:00,  1.05it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","curl-7.86.0          | 87 KB     | :  18% 0.18329902443390317/1 [00:00<00:00,  1.27it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","libssh2-1.10.0       | 274 KB    | :   6% 0.058380219709737995/1 [00:00<00:02,  2.49s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","libedit-3.1.20221030 | 181 KB    | : 100% 1.0/1 [00:00<00:00,  6.04it/s]               \u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","\n","krb5-1.19.2          | 1.2 MB    | :   1% 0.012786055208538188/1 [00:00<00:13, 13.56s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","libcurl-7.86.0       | 368 KB    | : 100% 1.0/1 [00:00<00:00,  4.16it/s]\n","\n","\n","\n","\n","libev-4.33           | 111 KB    | : 100% 1.0/1 [00:00<00:00,  4.30it/s]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","libev-4.33           | 111 KB    | : 100% 1.0/1 [00:00<00:00,  4.30it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","libnghttp2-1.46.0    | 680 KB    | : 100% 1.0/1 [00:00<00:00,  3.50it/s]                \u001b[A\n","libnghttp2-1.46.0    | 680 KB    | : 100% 1.0/1 [00:00<00:00,  3.50it/s]\u001b[A\n","\n","\n","\n","\n","\n","curl-7.86.0          | 87 KB     | : 100% 1.0/1 [00:00<00:00,  3.42it/s]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","curl-7.86.0          | 87 KB     | : 100% 1.0/1 [00:00<00:00,  3.42it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","libssh2-1.10.0       | 274 KB    | : 100% 1.0/1 [00:00<00:00,  3.13it/s]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","libssh2-1.10.0       | 274 KB    | : 100% 1.0/1 [00:00<00:00,  3.13it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","samtools-1.3.1       | 470 KB    | :   3% 0.0340255149827318/1 [00:00<00:10, 10.79s/it]\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","\n","krb5-1.19.2          | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.70it/s]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","\n","krb5-1.19.2          | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.70it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","samtools-1.3.1       | 470 KB    | : 100% 1.0/1 [00:00<00:00,  2.42it/s]               \u001b[A\u001b[A\n","\n","                                                                        \n","                                                                        \u001b[A\n","\n","                                                                        \u001b[A\u001b[A\n","\n","\n","                                                                        \u001b[A\u001b[A\u001b[A\n","\n","\n","\n","                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","\n","\n","\n","\n","\n","\n","\n","                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n","Preparing transaction: \\ \b\bdone\n","Verifying transaction: / \b\b- \b\bdone\n","Executing transaction: | \b\bdone\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed with initial frozen solve. Retrying with flexible solve.\n","Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n","Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bfailed with initial frozen solve. Retrying with flexible solve.\n","Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \n","Found conflicts! Looking for incompatible packages.\n","This can take several minutes.  Press CTRL-C to abort.\n","\b\bfailed\n","\n","UnsatisfiableError: The following specifications were found\n","to be incompatible with the existing python installation in your environment:\n","\n","Specifications:\n","\n","  - htseq -> python[version='<3.0.0']\n","\n","Your python: python=3.10\n","\n","If python is on the left-most side of the chain, that's the version you've asked for.\n","When python appears to the right, that indicates that the thing on the left is somehow\n","not available for the python version you are constrained to. Note that conda will not\n","change your python version to a different minor version unless you explicitly specify\n","that.\n","\n","The following specifications were found to be incompatible with your system:\n","\n","  - feature:/linux-64::__glibc==2.27=0\n","  - feature:|@/linux-64::__glibc==2.27=0\n","  - htseq -> libgcc-ng[version='>=10.3.0'] -> __glibc[version='>=2.17']\n","\n","Your installed version is: 2.27\n","\n","\n"]}]},{"cell_type":"code","source":["!STAR"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"RwF_0dYSCnFM","executionInfo":{"status":"ok","timestamp":1673528368305,"user_tz":-120,"elapsed":284,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"cb56c19b-2f9f-4be6-91f4-a97766f45666"},"execution_count":22,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","EXITING because of fatal input ERROR: could not open readFilesIn=Read1\n","\n","Jan 12 12:59:27 ...... FATAL ERROR, exiting\n"]}]},{"cell_type":"code","source":["%%bash \n","cd datasets/\n","fastqc *.fastq\n","cd ..\n","\n","mkdir trim/\n","cd datasets/\n","mv *.fastq ../trim/\n"],"metadata":{"id":"fM6Z6ylvu3xe","executionInfo":{"status":"ok","timestamp":1673528656988,"user_tz":-120,"elapsed":21739,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"97702cca-731a-4f19-922f-25e1cc328d8b"},"execution_count":24,"outputs":[{"output_type":"stream","name":"stdout","text":["Analysis complete for brain1_R1.fastq\n","Analysis complete for brain1_R2.fastq\n","Analysis complete for brain2_R1.fastq\n","Analysis complete for brain2_R2.fastq\n","Analysis complete for uhr1_R1.fastq\n","Analysis complete for uhr1_R2.fastq\n","Analysis complete for uhr2_R1.fastq\n","Analysis complete for uhr2_R2.fastq\n"]},{"output_type":"stream","name":"stderr","text":["Started analysis of brain1_R1.fastq\n","Approx 5% complete for brain1_R1.fastq\n","Approx 10% complete for brain1_R1.fastq\n","Approx 15% complete for brain1_R1.fastq\n","Approx 20% complete for brain1_R1.fastq\n","Approx 25% complete for brain1_R1.fastq\n","Approx 30% complete for brain1_R1.fastq\n","Approx 35% complete for brain1_R1.fastq\n","Approx 40% complete for brain1_R1.fastq\n","Approx 45% complete for brain1_R1.fastq\n","Approx 50% complete for brain1_R1.fastq\n","Approx 55% complete for brain1_R1.fastq\n","Approx 60% complete for brain1_R1.fastq\n","Approx 65% complete for brain1_R1.fastq\n","Approx 70% complete for brain1_R1.fastq\n","Approx 75% complete for brain1_R1.fastq\n","Approx 80% complete for brain1_R1.fastq\n","Approx 85% complete for brain1_R1.fastq\n","Approx 90% complete for brain1_R1.fastq\n","Approx 95% complete for brain1_R1.fastq\n","Started analysis of brain1_R2.fastq\n","Approx 5% complete for brain1_R2.fastq\n","Approx 10% complete for brain1_R2.fastq\n","Approx 15% complete for brain1_R2.fastq\n","Approx 20% complete for brain1_R2.fastq\n","Approx 25% complete for brain1_R2.fastq\n","Approx 30% complete for brain1_R2.fastq\n","Approx 35% complete for brain1_R2.fastq\n","Approx 40% complete for brain1_R2.fastq\n","Approx 45% complete for brain1_R2.fastq\n","Approx 50% complete for brain1_R2.fastq\n","Approx 55% complete for brain1_R2.fastq\n","Approx 60% complete for brain1_R2.fastq\n","Approx 65% complete for brain1_R2.fastq\n","Approx 70% complete for brain1_R2.fastq\n","Approx 75% complete for brain1_R2.fastq\n","Approx 80% complete for brain1_R2.fastq\n","Approx 85% complete for brain1_R2.fastq\n","Approx 90% complete for brain1_R2.fastq\n","Approx 95% complete for brain1_R2.fastq\n","Started analysis of brain2_R1.fastq\n","Approx 5% complete for brain2_R1.fastq\n","Approx 10% complete for brain2_R1.fastq\n","Approx 15% complete for brain2_R1.fastq\n","Approx 20% complete for brain2_R1.fastq\n","Approx 25% complete for brain2_R1.fastq\n","Approx 30% complete for brain2_R1.fastq\n","Approx 35% complete for brain2_R1.fastq\n","Approx 40% complete for brain2_R1.fastq\n","Approx 45% complete for brain2_R1.fastq\n","Approx 50% complete for brain2_R1.fastq\n","Approx 55% complete for brain2_R1.fastq\n","Approx 60% complete for brain2_R1.fastq\n","Approx 65% complete for brain2_R1.fastq\n","Approx 70% complete for brain2_R1.fastq\n","Approx 75% complete for brain2_R1.fastq\n","Approx 80% complete for brain2_R1.fastq\n","Approx 85% complete for brain2_R1.fastq\n","Approx 90% complete for brain2_R1.fastq\n","Approx 95% complete for brain2_R1.fastq\n","Started analysis of brain2_R2.fastq\n","Approx 5% complete for brain2_R2.fastq\n","Approx 10% complete for brain2_R2.fastq\n","Approx 15% complete for brain2_R2.fastq\n","Approx 20% complete for brain2_R2.fastq\n","Approx 25% complete for brain2_R2.fastq\n","Approx 30% complete for brain2_R2.fastq\n","Approx 35% complete for brain2_R2.fastq\n","Approx 40% complete for brain2_R2.fastq\n","Approx 45% complete for brain2_R2.fastq\n","Approx 50% complete for brain2_R2.fastq\n","Approx 55% complete for brain2_R2.fastq\n","Approx 60% complete for brain2_R2.fastq\n","Approx 65% complete for brain2_R2.fastq\n","Approx 70% complete for brain2_R2.fastq\n","Approx 75% complete for brain2_R2.fastq\n","Approx 80% complete for brain2_R2.fastq\n","Approx 85% complete for brain2_R2.fastq\n","Approx 90% complete for brain2_R2.fastq\n","Approx 95% complete for brain2_R2.fastq\n","Started analysis of uhr1_R1.fastq\n","Approx 5% complete for uhr1_R1.fastq\n","Approx 10% complete for uhr1_R1.fastq\n","Approx 15% complete for uhr1_R1.fastq\n","Approx 20% complete for uhr1_R1.fastq\n","Approx 25% complete for uhr1_R1.fastq\n","Approx 30% complete for uhr1_R1.fastq\n","Approx 35% complete for uhr1_R1.fastq\n","Approx 40% complete for uhr1_R1.fastq\n","Approx 45% complete for uhr1_R1.fastq\n","Approx 50% complete for uhr1_R1.fastq\n","Approx 55% complete for uhr1_R1.fastq\n","Approx 60% complete for uhr1_R1.fastq\n","Approx 65% complete for uhr1_R1.fastq\n","Approx 70% complete for uhr1_R1.fastq\n","Approx 75% complete for uhr1_R1.fastq\n","Approx 80% complete for uhr1_R1.fastq\n","Approx 85% complete for uhr1_R1.fastq\n","Approx 90% complete for uhr1_R1.fastq\n","Approx 95% complete for uhr1_R1.fastq\n","Started analysis of uhr1_R2.fastq\n","Approx 5% complete for uhr1_R2.fastq\n","Approx 10% complete for uhr1_R2.fastq\n","Approx 15% complete for uhr1_R2.fastq\n","Approx 20% complete for uhr1_R2.fastq\n","Approx 25% complete for uhr1_R2.fastq\n","Approx 30% complete for uhr1_R2.fastq\n","Approx 35% complete for uhr1_R2.fastq\n","Approx 40% complete for uhr1_R2.fastq\n","Approx 45% complete for uhr1_R2.fastq\n","Approx 50% complete for uhr1_R2.fastq\n","Approx 55% complete for uhr1_R2.fastq\n","Approx 60% complete for uhr1_R2.fastq\n","Approx 65% complete for uhr1_R2.fastq\n","Approx 70% complete for uhr1_R2.fastq\n","Approx 75% complete for uhr1_R2.fastq\n","Approx 80% complete for uhr1_R2.fastq\n","Approx 85% complete for uhr1_R2.fastq\n","Approx 90% complete for uhr1_R2.fastq\n","Approx 95% complete for uhr1_R2.fastq\n","Started analysis of uhr2_R1.fastq\n","Approx 5% complete for uhr2_R1.fastq\n","Approx 10% complete for uhr2_R1.fastq\n","Approx 15% complete for uhr2_R1.fastq\n","Approx 20% complete for uhr2_R1.fastq\n","Approx 25% complete for uhr2_R1.fastq\n","Approx 30% complete for uhr2_R1.fastq\n","Approx 35% complete for uhr2_R1.fastq\n","Approx 40% complete for uhr2_R1.fastq\n","Approx 45% complete for uhr2_R1.fastq\n","Approx 50% complete for uhr2_R1.fastq\n","Approx 55% complete for uhr2_R1.fastq\n","Approx 60% complete for uhr2_R1.fastq\n","Approx 65% complete for uhr2_R1.fastq\n","Approx 70% complete for uhr2_R1.fastq\n","Approx 75% complete for uhr2_R1.fastq\n","Approx 80% complete for uhr2_R1.fastq\n","Approx 85% complete for uhr2_R1.fastq\n","Approx 90% complete for uhr2_R1.fastq\n","Approx 95% complete for uhr2_R1.fastq\n","Started analysis of uhr2_R2.fastq\n","Approx 5% complete for uhr2_R2.fastq\n","Approx 10% complete for uhr2_R2.fastq\n","Approx 15% complete for uhr2_R2.fastq\n","Approx 20% complete for uhr2_R2.fastq\n","Approx 25% complete for uhr2_R2.fastq\n","Approx 30% complete for uhr2_R2.fastq\n","Approx 35% complete for uhr2_R2.fastq\n","Approx 40% complete for uhr2_R2.fastq\n","Approx 45% complete for uhr2_R2.fastq\n","Approx 50% complete for uhr2_R2.fastq\n","Approx 55% complete for uhr2_R2.fastq\n","Approx 60% complete for uhr2_R2.fastq\n","Approx 65% complete for uhr2_R2.fastq\n","Approx 70% complete for uhr2_R2.fastq\n","Approx 75% complete for uhr2_R2.fastq\n","Approx 80% complete for uhr2_R2.fastq\n","Approx 85% complete for uhr2_R2.fastq\n","Approx 90% complete for uhr2_R2.fastq\n","Approx 95% complete for uhr2_R2.fastq\n"]}]},{"cell_type":"markdown","source":["run trimmomatic on obtained data"],"metadata":{"id":"ZTRi3DinyZkr"}},{"cell_type":"code","source":["'''\n","!trimmomatic PE trim/brain1_R1.fastq trim/brain1_R2.fastq trim/brain1_R1.trim.fastq \\\n","trim/brain1_SE1.trim.fastq trim/brain1_R2.trim.fastq trim/brain1_SE2.trim.fastq \\\n","ILLUMINACLIP:adaptor.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50 2> trim/brain1.trim.log\n","'''\n","!trimmomatic PE trim/brain2_R1.fastq trim/brain2_R2.fastq trim/brain2_R1.trim.fastq \\\n","trim/brain2_SE1.trim.fastq trim/brain2_R2.trim.fastq trim/brain2_SE2.trim.fastq \\\n","ILLUMINACLIP:adaptor.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50 2> trim/brain2.trim.log\n","\n","!trimmomatic PE trim/uhr1_R1.fastq trim/uhr1_R2.fastq trim/uhr1_R1.trim.fastq \\\n","trim/uhr1_SE1.trim.fastq trim/uhr1_R2.trim.fastq trim/uhr1_SE2.trim.fastq \\\n","ILLUMINACLIP:adaptor.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50 2> trim/uhr1.trim.log\n","\n","!trimmomatic PE trim/uhr2_R1.fastq trim/uhr2_R2.fastq trim/uhr2_R1.trim.fastq \\\n","trim/uhr2_SE1.trim.fastq trim/uhr2_R2.trim.fastq trim/uhr2_SE2.trim.fastq \\\n","ILLUMINACLIP:adaptor.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50 2> trim/uhr2.trim.log "],"metadata":{"id":"Bi1DfhmSwU-C","executionInfo":{"status":"ok","timestamp":1673530330799,"user_tz":-120,"elapsed":12222,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":26,"outputs":[]},{"cell_type":"markdown","source":["run fastqc for all generated files"],"metadata":{"id":"tUX6M6pwyVmc"}},{"cell_type":"code","source":["%%bash\n","cd trim/\n","fastqc *.trim.fastq"],"metadata":{"id":"TeaoWUNJxtzN","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673530742598,"user_tz":-120,"elapsed":43834,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"321f2923-04f5-408f-b27e-15981f392f9f"},"execution_count":27,"outputs":[{"output_type":"stream","name":"stdout","text":["Analysis complete for brain1_R1.trim.fastq\n","Analysis complete for brain1_R2.trim.fastq\n","Analysis complete for brain1_SE1.trim.fastq\n","Analysis complete for brain1_SE2.trim.fastq\n","Analysis complete for brain2_R1.trim.fastq\n","Analysis complete for brain2_R2.trim.fastq\n","Analysis complete for brain2_SE1.trim.fastq\n","Analysis complete for brain2_SE2.trim.fastq\n","Analysis complete for uhr1_R1.trim.fastq\n","Analysis complete for uhr1_R2.trim.fastq\n","Analysis complete for uhr1_SE1.trim.fastq\n","Analysis complete for uhr1_SE2.trim.fastq\n","Analysis complete for uhr2_R1.trim.fastq\n","Analysis complete for uhr2_R2.trim.fastq\n","Analysis complete for uhr2_SE1.trim.fastq\n","Analysis complete for uhr2_SE2.trim.fastq\n"]},{"output_type":"stream","name":"stderr","text":["Started analysis of brain1_R1.trim.fastq\n","Approx 5% complete for brain1_R1.trim.fastq\n","Approx 10% complete for brain1_R1.trim.fastq\n","Approx 15% complete for brain1_R1.trim.fastq\n","Approx 20% complete for brain1_R1.trim.fastq\n","Approx 25% complete for brain1_R1.trim.fastq\n","Approx 30% complete for brain1_R1.trim.fastq\n","Approx 35% complete for brain1_R1.trim.fastq\n","Approx 40% complete for brain1_R1.trim.fastq\n","Approx 45% complete for brain1_R1.trim.fastq\n","Approx 50% complete for brain1_R1.trim.fastq\n","Approx 55% complete for brain1_R1.trim.fastq\n","Approx 60% complete for brain1_R1.trim.fastq\n","Approx 65% complete for brain1_R1.trim.fastq\n","Approx 70% complete for brain1_R1.trim.fastq\n","Approx 75% complete for brain1_R1.trim.fastq\n","Approx 80% complete for brain1_R1.trim.fastq\n","Approx 85% complete for brain1_R1.trim.fastq\n","Approx 90% complete for brain1_R1.trim.fastq\n","Approx 95% complete for brain1_R1.trim.fastq\n","Started analysis of brain1_R2.trim.fastq\n","Approx 5% complete for brain1_R2.trim.fastq\n","Approx 10% complete for brain1_R2.trim.fastq\n","Approx 15% complete for brain1_R2.trim.fastq\n","Approx 20% complete for brain1_R2.trim.fastq\n","Approx 25% complete for brain1_R2.trim.fastq\n","Approx 30% complete for brain1_R2.trim.fastq\n","Approx 35% complete for brain1_R2.trim.fastq\n","Approx 40% complete for brain1_R2.trim.fastq\n","Approx 45% complete for brain1_R2.trim.fastq\n","Approx 50% complete for brain1_R2.trim.fastq\n","Approx 55% complete for brain1_R2.trim.fastq\n","Approx 60% complete for brain1_R2.trim.fastq\n","Approx 65% complete for brain1_R2.trim.fastq\n","Approx 70% complete for brain1_R2.trim.fastq\n","Approx 75% complete for brain1_R2.trim.fastq\n","Approx 80% complete for brain1_R2.trim.fastq\n","Approx 85% complete for brain1_R2.trim.fastq\n","Approx 90% complete for brain1_R2.trim.fastq\n","Approx 95% complete for brain1_R2.trim.fastq\n","Started analysis of brain1_SE1.trim.fastq\n","Approx 30% complete for brain1_SE1.trim.fastq\n","Approx 60% complete for brain1_SE1.trim.fastq\n","Approx 90% complete for brain1_SE1.trim.fastq\n","Started analysis of brain1_SE2.trim.fastq\n","Started analysis of brain2_R1.trim.fastq\n","Approx 5% complete for brain2_R1.trim.fastq\n","Approx 10% complete for brain2_R1.trim.fastq\n","Approx 15% complete for brain2_R1.trim.fastq\n","Approx 20% complete for brain2_R1.trim.fastq\n","Approx 25% complete for brain2_R1.trim.fastq\n","Approx 30% complete for brain2_R1.trim.fastq\n","Approx 35% complete for brain2_R1.trim.fastq\n","Approx 40% complete for brain2_R1.trim.fastq\n","Approx 45% complete for brain2_R1.trim.fastq\n","Approx 50% complete for brain2_R1.trim.fastq\n","Approx 55% complete for brain2_R1.trim.fastq\n","Approx 60% complete for brain2_R1.trim.fastq\n","Approx 65% complete for brain2_R1.trim.fastq\n","Approx 70% complete for brain2_R1.trim.fastq\n","Approx 75% complete for brain2_R1.trim.fastq\n","Approx 80% complete for brain2_R1.trim.fastq\n","Approx 85% complete for brain2_R1.trim.fastq\n","Approx 90% complete for brain2_R1.trim.fastq\n","Approx 95% complete for brain2_R1.trim.fastq\n","Started analysis of brain2_R2.trim.fastq\n","Approx 5% complete for brain2_R2.trim.fastq\n","Approx 10% complete for brain2_R2.trim.fastq\n","Approx 15% complete for brain2_R2.trim.fastq\n","Approx 20% complete for brain2_R2.trim.fastq\n","Approx 25% complete for brain2_R2.trim.fastq\n","Approx 30% complete for brain2_R2.trim.fastq\n","Approx 35% complete for brain2_R2.trim.fastq\n","Approx 40% complete for brain2_R2.trim.fastq\n","Approx 45% complete for brain2_R2.trim.fastq\n","Approx 50% complete for brain2_R2.trim.fastq\n","Approx 55% complete for brain2_R2.trim.fastq\n","Approx 60% complete for brain2_R2.trim.fastq\n","Approx 65% complete for brain2_R2.trim.fastq\n","Approx 70% complete for brain2_R2.trim.fastq\n","Approx 75% complete for brain2_R2.trim.fastq\n","Approx 80% complete for brain2_R2.trim.fastq\n","Approx 85% complete for brain2_R2.trim.fastq\n","Approx 90% complete for brain2_R2.trim.fastq\n","Approx 95% complete for brain2_R2.trim.fastq\n","Started analysis of brain2_SE1.trim.fastq\n","Approx 30% complete for brain2_SE1.trim.fastq\n","Approx 65% complete for brain2_SE1.trim.fastq\n","Started analysis of brain2_SE2.trim.fastq\n","Started analysis of uhr1_R1.trim.fastq\n","Approx 5% complete for uhr1_R1.trim.fastq\n","Approx 10% complete for uhr1_R1.trim.fastq\n","Approx 15% complete for uhr1_R1.trim.fastq\n","Approx 20% complete for uhr1_R1.trim.fastq\n","Approx 25% complete for uhr1_R1.trim.fastq\n","Approx 30% complete for uhr1_R1.trim.fastq\n","Approx 35% complete for uhr1_R1.trim.fastq\n","Approx 40% complete for uhr1_R1.trim.fastq\n","Approx 45% complete for uhr1_R1.trim.fastq\n","Approx 50% complete for uhr1_R1.trim.fastq\n","Approx 55% complete for uhr1_R1.trim.fastq\n","Approx 60% complete for uhr1_R1.trim.fastq\n","Approx 65% complete for uhr1_R1.trim.fastq\n","Approx 70% complete for uhr1_R1.trim.fastq\n","Approx 75% complete for uhr1_R1.trim.fastq\n","Approx 80% complete for uhr1_R1.trim.fastq\n","Approx 85% complete for uhr1_R1.trim.fastq\n","Approx 90% complete for uhr1_R1.trim.fastq\n","Approx 95% complete for uhr1_R1.trim.fastq\n","Started analysis of uhr1_R2.trim.fastq\n","Approx 5% complete for uhr1_R2.trim.fastq\n","Approx 10% complete for uhr1_R2.trim.fastq\n","Approx 15% complete for uhr1_R2.trim.fastq\n","Approx 20% complete for uhr1_R2.trim.fastq\n","Approx 25% complete for uhr1_R2.trim.fastq\n","Approx 30% complete for uhr1_R2.trim.fastq\n","Approx 35% complete for uhr1_R2.trim.fastq\n","Approx 40% complete for uhr1_R2.trim.fastq\n","Approx 45% complete for uhr1_R2.trim.fastq\n","Approx 50% complete for uhr1_R2.trim.fastq\n","Approx 55% complete for uhr1_R2.trim.fastq\n","Approx 60% complete for uhr1_R2.trim.fastq\n","Approx 65% complete for uhr1_R2.trim.fastq\n","Approx 70% complete for uhr1_R2.trim.fastq\n","Approx 75% complete for uhr1_R2.trim.fastq\n","Approx 80% complete for uhr1_R2.trim.fastq\n","Approx 85% complete for uhr1_R2.trim.fastq\n","Approx 90% complete for uhr1_R2.trim.fastq\n","Approx 95% complete for uhr1_R2.trim.fastq\n","Started analysis of uhr1_SE1.trim.fastq\n","Approx 20% complete for uhr1_SE1.trim.fastq\n","Approx 40% complete for uhr1_SE1.trim.fastq\n","Approx 55% complete for uhr1_SE1.trim.fastq\n","Approx 75% complete for uhr1_SE1.trim.fastq\n","Approx 95% complete for uhr1_SE1.trim.fastq\n","Started analysis of uhr1_SE2.trim.fastq\n","Started analysis of uhr2_R1.trim.fastq\n","Approx 5% complete for uhr2_R1.trim.fastq\n","Approx 10% complete for uhr2_R1.trim.fastq\n","Approx 15% complete for uhr2_R1.trim.fastq\n","Approx 20% complete for uhr2_R1.trim.fastq\n","Approx 25% complete for uhr2_R1.trim.fastq\n","Approx 30% complete for uhr2_R1.trim.fastq\n","Approx 35% complete for uhr2_R1.trim.fastq\n","Approx 40% complete for uhr2_R1.trim.fastq\n","Approx 45% complete for uhr2_R1.trim.fastq\n","Approx 50% complete for uhr2_R1.trim.fastq\n","Approx 55% complete for uhr2_R1.trim.fastq\n","Approx 60% complete for uhr2_R1.trim.fastq\n","Approx 65% complete for uhr2_R1.trim.fastq\n","Approx 70% complete for uhr2_R1.trim.fastq\n","Approx 75% complete for uhr2_R1.trim.fastq\n","Approx 80% complete for uhr2_R1.trim.fastq\n","Approx 85% complete for uhr2_R1.trim.fastq\n","Approx 90% complete for uhr2_R1.trim.fastq\n","Approx 95% complete for uhr2_R1.trim.fastq\n","Started analysis of uhr2_R2.trim.fastq\n","Approx 5% complete for uhr2_R2.trim.fastq\n","Approx 10% complete for uhr2_R2.trim.fastq\n","Approx 15% complete for uhr2_R2.trim.fastq\n","Approx 20% complete for uhr2_R2.trim.fastq\n","Approx 25% complete for uhr2_R2.trim.fastq\n","Approx 30% complete for uhr2_R2.trim.fastq\n","Approx 35% complete for uhr2_R2.trim.fastq\n","Approx 40% complete for uhr2_R2.trim.fastq\n","Approx 45% complete for uhr2_R2.trim.fastq\n","Approx 50% complete for uhr2_R2.trim.fastq\n","Approx 55% complete for uhr2_R2.trim.fastq\n","Approx 60% complete for uhr2_R2.trim.fastq\n","Approx 65% complete for uhr2_R2.trim.fastq\n","Approx 70% complete for uhr2_R2.trim.fastq\n","Approx 75% complete for uhr2_R2.trim.fastq\n","Approx 80% complete for uhr2_R2.trim.fastq\n","Approx 85% complete for uhr2_R2.trim.fastq\n","Approx 90% complete for uhr2_R2.trim.fastq\n","Approx 95% complete for uhr2_R2.trim.fastq\n","Started analysis of uhr2_SE1.trim.fastq\n","Approx 25% complete for uhr2_SE1.trim.fastq\n","Approx 55% complete for uhr2_SE1.trim.fastq\n","Approx 80% complete for uhr2_SE1.trim.fastq\n","Started analysis of uhr2_SE2.trim.fastq\n"]}]},{"cell_type":"markdown","source":["create new directory for each alignment"],"metadata":{"id":"Ii2hhUQnyPUU"}},{"cell_type":"code","source":["%%bash\n","mkdir genome\n","mkdir gtf\n","mkdir GRCh38.108.chrom22\n","mkdir brain1 brain2 uhr1 uhr2\n","#move chromosome 22 sequence to genome directory \n","mv chr22.fa genome/"],"metadata":{"id":"X87BwitFx4gV","executionInfo":{"status":"ok","timestamp":1673531680383,"user_tz":-120,"elapsed":290,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":28,"outputs":[]},{"cell_type":"markdown","source":["get the genome feature file"],"metadata":{"id":"6fh0GWgQzZbq"}},{"cell_type":"code","source":["#!wget https://ftp.ensembl.org/pub/release-108/gtf/homo_sapiens/Homo_sapiens.GRCh38.108.gtf.gz\n","!gzip -d Homo_sapiens.GRCh38.108.gtf.gz"],"metadata":{"id":"isFPWqT3yOql","executionInfo":{"status":"ok","timestamp":1673535987850,"user_tz":-120,"elapsed":10337,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":31,"outputs":[]},{"cell_type":"markdown","source":["extract features of chr22 from the whole genome feature file then move the file containing extracted features to its directory"],"metadata":{"id":"MfWMcAzTzdZl"}},{"cell_type":"code","source":["!grep -P '^22\\t' Homo_sapiens.GRCh38.108.gtf > chr22.gtf\n","!mv chr22.gtf gtf/"],"metadata":{"id":"ZDyxbgJazc8q","executionInfo":{"status":"ok","timestamp":1673536026011,"user_tz":-120,"elapsed":2486,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":33,"outputs":[]},{"cell_type":"markdown","source":["Generate Reference Genome Index"],"metadata":{"id":"DHXsFCwC2b8S"}},{"cell_type":"code","source":["!STAR --runMode genomeGenerate --genomeDir GRCh38.108.chrom22/ \\\n","--genomeFastaFiles genome/chr22.fa --sjdbGTFfile gtf/chr22.gtf \\\n","--sjdbOverhang 75"],"metadata":{"id":"WW6YABkF0SRN","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673536143948,"user_tz":-120,"elapsed":94688,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"08430b56-038d-4f44-a5a2-3c493d8793d4"},"execution_count":34,"outputs":[{"output_type":"stream","name":"stdout","text":["Jan 12 15:07:28 ..... Started STAR run\n","Jan 12 15:07:28 ... Starting to generate Genome files\n","Jan 12 15:07:30 ... starting to sort  Suffix Array. This may take a long time...\n","Jan 12 15:07:30 ... sorting Suffix Array chunks and saving them to disk...\n","Jan 12 15:08:14 ... loading chunks from disk, packing SA...\n","Jan 12 15:08:15 ... Finished generating suffix array\n","Jan 12 15:08:15 ... starting to generate Suffix Array index...\n","Jan 12 15:08:36 ..... Processing annotations GTF\n","Jan 12 15:08:36 ..... Inserting junctions into the genome indices\n","Jan 12 15:08:54 ... writing Genome to disk ...\n","Jan 12 15:08:54 ... writing Suffix Array to disk ...\n","Jan 12 15:08:55 ... writing SAindex to disk\n","Jan 12 15:09:03 ..... Finished successfully\n"]}]},{"cell_type":"markdown","source":["**Reads Alignment**"],"metadata":{"id":"_9zPSw-02pHV"}},{"cell_type":"code","source":["# Align trimmed reads\n","!STAR --runThreadN 2 --genomeDir GRCh38.108.chrom22 --readFilesIn trim/brain1_R1.trim.fastq trim/brain1_R2.trim.fastq --outFileNamePrefix brain1/\n","!STAR --runThreadN 2 --genomeDir GRCh38.108.chrom22 --readFilesIn trim/brain2_R1.trim.fastq trim/brain2_R2.trim.fastq --outFileNamePrefix brain2/\n","!STAR --runThreadN 2 --genomeDir GRCh38.108.chrom22 --readFilesIn trim/uhr1_R1.trim.fastq trim/uhr1_R2.trim.fastq --outFileNamePrefix uhr1/\n","!STAR --runThreadN 2 --genomeDir GRCh38.108.chrom22 --readFilesIn trim/uhr2_R1.trim.fastq trim/uhr2_R2.trim.fastq --outFileNamePrefix uhr2/"],"metadata":{"id":"YA94KjRR2onT","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673536316535,"user_tz":-120,"elapsed":139591,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"140a74e7-470f-40cf-b158-481b0ffc4288"},"execution_count":35,"outputs":[{"output_type":"stream","name":"stdout","text":["Jan 12 15:09:36 ..... Started STAR run\n","Jan 12 15:09:36 ..... Loading genome\n","Jan 12 15:09:37 ..... Started mapping\n","Jan 12 15:10:04 ..... Finished successfully\n","Jan 12 15:10:04 ..... Started STAR run\n","Jan 12 15:10:04 ..... Loading genome\n","Jan 12 15:10:05 ..... Started mapping\n","Jan 12 15:10:26 ..... Finished successfully\n","Jan 12 15:10:26 ..... Started STAR run\n","Jan 12 15:10:26 ..... Loading genome\n","Jan 12 15:10:27 ..... Started mapping\n","Jan 12 15:11:19 ..... Finished successfully\n","Jan 12 15:11:19 ..... Started STAR run\n","Jan 12 15:11:19 ..... Loading genome\n","Jan 12 15:11:21 ..... Started mapping\n","Jan 12 15:11:55 ..... Finished successfully\n"]}]},{"cell_type":"markdown","source":["creat a new directory where the BAM files will be generated"],"metadata":{"id":"MRpypPbQ5HUI"}},{"cell_type":"code","source":["%%bash\n","mkdir star_SAM\n","cd star_SAM\n","#link the sam files to virtual files in this directory (similar to shortcut in windows)\n","ln -s ../brain1/Aligned.out.sam brain1.sam\n","ln -s ../brain2/Aligned.out.sam brain2.sam\n","ln -s ../uhr1/Aligned.out.sam uhr1.sam\n","ln -s ../uhr2/Aligned.out.sam uhr2.sam"],"metadata":{"id":"5XCuw5pd3W46","executionInfo":{"status":"ok","timestamp":1673536326370,"user_tz":-120,"elapsed":280,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":36,"outputs":[]},{"cell_type":"markdown","source":["sort and convert the SAM files to BAM"],"metadata":{"id":"WCbyedLw5D5C"}},{"cell_type":"code","source":["!samtools sort star_SAM/brain1.sam -o star_SAM/brain1.bam\n","!samtools sort star_SAM/brain2.sam -o star_SAM/brain2.bam\n","!samtools sort star_SAM/uhr1.sam -o star_SAM/uhr1.bam\n","!samtools sort star_SAM/uhr2.sam -o star_SAM/uhr2.bam"],"metadata":{"id":"4kUXD0Eg4j3T","executionInfo":{"status":"ok","timestamp":1673536381159,"user_tz":-120,"elapsed":9887,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":37,"outputs":[]},{"cell_type":"markdown","source":["**Differential expression analysis**\n","\n","1- Count-based Gene Expression Using HTSeq–count & edgeR"],"metadata":{"id":"y6fWdNkb5it4"}},{"cell_type":"code","source":["!conda install -c bioconda htseq"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"9RzFSf4KiyLq","executionInfo":{"status":"ok","timestamp":1673536525349,"user_tz":-120,"elapsed":82915,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"e0cd9d38-4691-43b1-8db6-f3d0a27cfb8b"},"execution_count":39,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bfailed with initial frozen solve. Retrying with flexible solve.\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n","Collecting package metadata (repodata.json): | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with initial frozen solve. Retrying with flexible solve.\n","Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \n","Found conflicts! Looking for incompatible packages.\n","This can take several minutes.  Press CTRL-C to abort.\n","\b\bfailed\n","\n","UnsatisfiableError: The following specifications were found\n","to be incompatible with the existing python installation in your environment:\n","\n","Specifications:\n","\n","  - htseq -> python[version='<3.0.0|>=3.11,<3.12.0a0']\n","\n","Your python: python=3.10\n","\n","If python is on the left-most side of the chain, that's the version you've asked for.\n","When python appears to the right, that indicates that the thing on the left is somehow\n","not available for the python version you are constrained to. Note that conda will not\n","change your python version to a different minor version unless you explicitly specify\n","that.\n","\n","The following specifications were found to be incompatible with your system:\n","\n","  - feature:/linux-64::__glibc==2.27=0\n","  - htseq -> libgcc-ng[version='>=10.3.0'] -> __glibc[version='>=2.17']\n","  - python=3.10 -> libgcc-ng[version='>=11.2.0'] -> __glibc[version='>=2.17']\n","\n","Your installed version is: 2.27\n","\n","\n"]}]},{"cell_type":"code","source":["!pip install Cython\n","!pip install pysam\n","!pip install HTSeq"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"xRDf2pgFjMr5","executionInfo":{"status":"ok","timestamp":1673536710605,"user_tz":-120,"elapsed":7448,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"3fa3ce0f-da90-422a-d8fc-aceab8211fd4"},"execution_count":44,"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting Cython\n","  Downloading Cython-0.29.33-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m22.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: Cython\n","Successfully installed Cython-0.29.33\n","\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n","\u001b[0mLooking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting pysam\n","  Downloading pysam-0.20.0-cp310-cp310-manylinux_2_24_x86_64.whl (15.5 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m15.5/15.5 MB\u001b[0m \u001b[31m15.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: pysam\n","Successfully installed pysam-0.20.0\n","\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n","\u001b[0mLooking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting HTSeq\n","  Downloading HTSeq-2.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.9 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m14.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/site-packages (from HTSeq) (1.24.1)\n","Requirement already satisfied: pysam in /usr/local/lib/python3.10/site-packages (from HTSeq) (0.20.0)\n","Installing collected packages: HTSeq\n","Successfully installed HTSeq-2.0.2\n","\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n","\u001b[0m"]}]},{"cell_type":"code","source":["!sudo apt-get install build-essential python3.6-dev python-numpy python-matplotlib python-pysam python-htseq"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Q5qm3SuJj9AZ","executionInfo":{"status":"ok","timestamp":1673536840844,"user_tz":-120,"elapsed":19249,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"e7f5e66e-b0af-45f0-f9c4-42cad6b01c19"},"execution_count":47,"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","build-essential is already the newest version (12.4ubuntu1).\n","python-numpy is already the newest version (1:1.13.3-2ubuntu1).\n","python-numpy set to manually installed.\n","python3.6-dev is already the newest version (3.6.9-1~18.04ubuntu1.9).\n","python3.6-dev set to manually installed.\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-460\n","Use 'sudo apt autoremove' to remove it.\n","The following additional packages will be installed:\n","  fonts-lyx javascript-common libhts2 libjs-jquery libjs-jquery-ui\n","  python-backports.functools-lru-cache python-cycler python-dateutil\n","  python-matplotlib-data python-olefile python-pil python-pyparsing python-six\n","  python-subprocess32 python-tz ttf-bitstream-vera\n","Suggested packages:\n","  apache2 | lighttpd | httpd libjs-jquery-ui-docs python-cycler-doc dvipng\n","  gir1.2-gtk-3.0 ghostscript inkscape ipython python-cairocffi\n","  python-configobj python-excelerator python-gobject python-matplotlib-doc\n","  python-nose python-qt4 python-scipy python-sip python-tornado python-traits\n","  python-wxgtk3.0 texlive-extra-utils texlive-latex-extra ttf-staypuft\n","  python-pil-doc python-pil-dbg python-pyparsing-doc\n","The following NEW packages will be installed:\n","  fonts-lyx javascript-common libhts2 libjs-jquery libjs-jquery-ui\n","  python-backports.functools-lru-cache python-cycler python-dateutil\n","  python-htseq python-matplotlib python-matplotlib-data python-olefile\n","  python-pil python-pyparsing python-pysam python-six python-subprocess32\n","  python-tz ttf-bitstream-vera\n","0 upgraded, 19 newly installed, 0 to remove and 21 not upgraded.\n","Need to get 11.2 MB of archives.\n","After this operation, 36.9 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 fonts-lyx all 2.2.4-0ubuntu0.18.04.1 [155 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 javascript-common all 11 [6,066 B]\n","Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 libjs-jquery all 3.2.1-1 [152 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-jquery-ui all 1.12.1+dfsg-5 [232 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-backports.functools-lru-cache all 1.4-2 [5,960 B]\n","Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cycler all 0.10.0-1 [7,520 B]\n","Get:9 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-dateutil all 2.6.1-1 [60.6 kB]\n","Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-htseq amd64 0.6.1p1-4build1 [203 kB]\n","Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 ttf-bitstream-vera all 1.10-8 [352 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-matplotlib-data all 2.1.1-2ubuntu3 [3,774 kB]\n","Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-tz all 2018.3-2 [31.6 kB]\n","Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-subprocess32 amd64 3.2.7-3 [27.2 kB]\n","Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-matplotlib amd64 2.1.1-2ubuntu3 [3,901 kB]\n","Get:17 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-olefile all 0.45.1-1 [33.2 kB]\n","Get:18 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-pil amd64 5.1.0-1ubuntu0.8 [303 kB]\n","Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-pysam amd64 0.14+ds-2 [1,615 kB]\n","Fetched 11.2 MB in 0s (24.1 MB/s)\n","debconf: unable to initialize frontend: Dialog\n","debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 19.)\n","debconf: falling back to frontend: Readline\n","debconf: unable to initialize frontend: Readline\n","debconf: (This frontend requires a controlling tty.)\n","debconf: falling back to frontend: Teletype\n","dpkg-preconfigure: unable to re-open stdin: \n","Selecting previously unselected package fonts-lyx.\n","(Reading database ... 124016 files and directories currently installed.)\n","Preparing to unpack .../00-fonts-lyx_2.2.4-0ubuntu0.18.04.1_all.deb ...\n","Unpacking fonts-lyx (2.2.4-0ubuntu0.18.04.1) ...\n","Selecting previously unselected package javascript-common.\n","Preparing to unpack .../01-javascript-common_11_all.deb ...\n","Unpacking javascript-common (11) ...\n","Selecting previously unselected package libhts2:amd64.\n","Preparing to unpack .../02-libhts2_1.7-2_amd64.deb ...\n","Unpacking libhts2:amd64 (1.7-2) ...\n","Selecting previously unselected package libjs-jquery.\n","Preparing to unpack .../03-libjs-jquery_3.2.1-1_all.deb ...\n","Unpacking libjs-jquery (3.2.1-1) ...\n","Selecting previously unselected package libjs-jquery-ui.\n","Preparing to unpack .../04-libjs-jquery-ui_1.12.1+dfsg-5_all.deb ...\n","Unpacking libjs-jquery-ui (1.12.1+dfsg-5) ...\n","Selecting previously unselected package python-backports.functools-lru-cache.\n","Preparing to unpack .../05-python-backports.functools-lru-cache_1.4-2_all.deb ...\n","Unpacking python-backports.functools-lru-cache (1.4-2) ...\n","Selecting previously unselected package python-six.\n","Preparing to unpack .../06-python-six_1.11.0-2_all.deb ...\n","Unpacking python-six (1.11.0-2) ...\n","Selecting previously unselected package python-cycler.\n","Preparing to unpack .../07-python-cycler_0.10.0-1_all.deb ...\n","Unpacking python-cycler (0.10.0-1) ...\n","Selecting previously unselected package python-dateutil.\n","Preparing to unpack .../08-python-dateutil_2.6.1-1_all.deb ...\n","Unpacking python-dateutil (2.6.1-1) ...\n","Selecting previously unselected package python-htseq.\n","Preparing to unpack .../09-python-htseq_0.6.1p1-4build1_amd64.deb ...\n","Unpacking python-htseq (0.6.1p1-4build1) ...\n","Selecting previously unselected package ttf-bitstream-vera.\n","Preparing to unpack .../10-ttf-bitstream-vera_1.10-8_all.deb ...\n","Unpacking ttf-bitstream-vera (1.10-8) ...\n","Selecting previously unselected package python-matplotlib-data.\n","Preparing to unpack .../11-python-matplotlib-data_2.1.1-2ubuntu3_all.deb ...\n","Unpacking python-matplotlib-data (2.1.1-2ubuntu3) ...\n","Selecting previously unselected package python-pyparsing.\n","Preparing to unpack .../12-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n","Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n","Selecting previously unselected package python-tz.\n","Preparing to unpack .../13-python-tz_2018.3-2_all.deb ...\n","Unpacking python-tz (2018.3-2) ...\n","Selecting previously unselected package python-subprocess32.\n","Preparing to unpack .../14-python-subprocess32_3.2.7-3_amd64.deb ...\n","Unpacking python-subprocess32 (3.2.7-3) ...\n","Selecting previously unselected package python-matplotlib.\n","Preparing to unpack .../15-python-matplotlib_2.1.1-2ubuntu3_amd64.deb ...\n","Unpacking python-matplotlib (2.1.1-2ubuntu3) ...\n","Selecting previously unselected package python-olefile.\n","Preparing to unpack .../16-python-olefile_0.45.1-1_all.deb ...\n","Unpacking python-olefile (0.45.1-1) ...\n","Selecting previously unselected package python-pil:amd64.\n","Preparing to unpack .../17-python-pil_5.1.0-1ubuntu0.8_amd64.deb ...\n","Unpacking python-pil:amd64 (5.1.0-1ubuntu0.8) ...\n","Selecting previously unselected package python-pysam.\n","Preparing to unpack .../18-python-pysam_0.14+ds-2_amd64.deb ...\n","Unpacking python-pysam (0.14+ds-2) ...\n","Setting up libjs-jquery (3.2.1-1) ...\n","Setting up python-backports.functools-lru-cache (1.4-2) ...\n","Setting up python-htseq (0.6.1p1-4build1) ...\n","Setting up python-tz (2018.3-2) ...\n","Setting up python-six (1.11.0-2) ...\n","Setting up libjs-jquery-ui (1.12.1+dfsg-5) ...\n","Setting up python-olefile (0.45.1-1) ...\n","Setting up ttf-bitstream-vera (1.10-8) ...\n","Setting up javascript-common (11) ...\n","Setting up python-subprocess32 (3.2.7-3) ...\n","Setting up python-pil:amd64 (5.1.0-1ubuntu0.8) ...\n","Setting up libhts2:amd64 (1.7-2) ...\n","Setting up fonts-lyx (2.2.4-0ubuntu0.18.04.1) ...\n","Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n","Setting up python-matplotlib-data (2.1.1-2ubuntu3) ...\n","Setting up python-pysam (0.14+ds-2) ...\n","Setting up python-cycler (0.10.0-1) ...\n","Setting up python-dateutil (2.6.1-1) ...\n","Setting up python-matplotlib (2.1.1-2ubuntu3) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n","Processing triggers for fontconfig (2.12.6-0ubuntu2) ...\n","Processing triggers for libc-bin (2.27-3ubuntu1.6) ...\n","/sbin/ldconfig.real: file /usr/local/lib/libjdwp.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libmanagement_agent.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjcef.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libawt_headless.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libzip.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libsunec.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjaas.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libsaproc.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnativewindow_awt.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libawt.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libverify.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnativewindow_drm.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libinstrument.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libfontmanager.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libcef.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libgluegen_rt.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjava.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libsplashscreen.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnewt_head.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjavajpeg.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libextnet.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/librmi.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libmlib_image.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libj2gss.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjsound.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libGLESv2.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjimage.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libawt_xawt.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libmanagement_ext.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libprefs.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libj2pcsc.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libunpack.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libattach.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/liblcms.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnewt_drm.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libdt_socket.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libmanagement.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libsctp.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnio.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjsig.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnativewindow_x11.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libj2pkcs11.so is truncated\n","\n","/sbin/ldconfig.real: /usr/local/lib/ is not a symbolic link\n","\n"]}]},{"cell_type":"code","source":["!htseq-count --format bam --order pos --mode intersection-strict --stranded reverse \\\n","--minaqual 1 --type exon --idattr gene_id star_SAM/brain1.bam gtf/chr22.gtf > brain1_gene_read_counts_table.tsv"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"9-pRB2pTkgcA","executionInfo":{"status":"ok","timestamp":1673536889201,"user_tz":-120,"elapsed":12951,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"145867e7-6354-4ffa-859f-9d72580bd616"},"execution_count":48,"outputs":[{"output_type":"stream","name":"stdout","text":["[E::idx_find_and_load] Could not retrieve index file for 'star_SAM/brain1.bam'\n","71454 GFF lines processed.\n","[E::idx_find_and_load] Could not retrieve index file for 'star_SAM/brain1.bam'\n","82847 alignment record pairs processed.\n"]}]},{"cell_type":"code","source":["#!htseq-count --format bam --order pos --mode intersection-strict --stranded reverse \\\n","#--minaqual 1 --type exon --idattr gene_id star_SAM/brain1.bam gtf/chr22.gtf > brain1_gene_read_counts_table.tsv\n","#!htseq-count --format bam --order pos --mode intersection-strict --stranded reverse \\\n","#--minaqual 1 --type exon --idattr gene_id star_SAM/brain2.bam gtf/chr22.gtf > brain2_gene_read_counts_table.tsv\n","#!htseq-count --format bam --order pos --mode intersection-strict --stranded reverse \\\n","#--minaqual 1 --type exon --idattr gene_id star_SAM/uhr1.bam gtf/chr22.gtf > uhr1_gene_read_counts_table.tsv\n","!htseq-count --format bam --order pos --mode intersection-strict --stranded reverse \\\n","--minaqual 1 --type exon --idattr gene_id star_SAM/uhr2.bam gtf/chr22.gtf > uhr2_gene_read_counts_table.tsv"],"metadata":{"id":"38SxsPpG5jKk","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673537991642,"user_tz":-120,"elapsed":14723,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"69a2c46f-3a21-472b-d2dc-0596956eabd4"},"execution_count":81,"outputs":[{"output_type":"stream","name":"stdout","text":["[E::idx_find_and_load] Could not retrieve index file for 'star_SAM/uhr2.bam'\n","71454 GFF lines processed.\n","[E::idx_find_and_load] Could not retrieve index file for 'star_SAM/uhr2.bam'\n","90337 alignment record pairs processed.\n"]}]},{"cell_type":"markdown","source":["The last 5 lines of the tsv file generated by HTSeq-count are\n","basic statistics. We can view them by using the tail command:"],"metadata":{"id":"HC-5xCZQ8NCJ"}},{"cell_type":"code","source":["!tail brain1_gene_read_counts_table.tsv"],"metadata":{"id":"xSuBnQ7j8Nzq","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673536981614,"user_tz":-120,"elapsed":377,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"900f7969-eab1-4cfa-b2cf-d33663bd5db3"},"execution_count":50,"outputs":[{"output_type":"stream","name":"stdout","text":["ENSG00000291064\t1\n","ENSG00000291085\t0\n","ENSG00000291087\t0\n","ENSG00000291240\t0\n","ENSG00000291299\t0\n","__no_feature\t16795\n","__ambiguous\t5698\n","__too_low_aQual\t0\n","__not_aligned\t0\n","__alignment_not_unique\t3032\n"]}]},{"cell_type":"markdown","source":["HTseq-count generates a 2-column file, the first column is for\n","\n","gene id and the second column is for the count of reads mapped to\n","the gene."],"metadata":{"id":"9pBlsEKm7eXt"}},{"cell_type":"code","source":["!head -5 brain1_gene_read_counts_table.tsv"],"metadata":{"id":"aJbhuXTC7fRi","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673537074703,"user_tz":-120,"elapsed":245,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"4b4b5c86-f029-4658-f8be-cae538322b95"},"execution_count":51,"outputs":[{"output_type":"stream","name":"stdout","text":["ENSG00000008735\t1920\n","ENSG00000015475\t0\n","ENSG00000025708\t0\n","ENSG00000025770\t0\n","ENSG00000040608\t0\n"]}]},{"cell_type":"markdown","source":["the file is sorted by the gene id.\n","\n","to sort it by the count reads use this:"],"metadata":{"id":"foKRN0ey7oKG"}},{"cell_type":"code","source":["!sort -k 2n brain1_gene_read_counts_table.tsv | sed '/^_/d' | head -5"],"metadata":{"id":"oIqcy-lo71RP","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1673537624934,"user_tz":-120,"elapsed":279,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}},"outputId":"e0d3874a-d36d-480d-8270-2f1340aafe6c"},"execution_count":69,"outputs":[{"output_type":"stream","name":"stdout","text":["ENSG00000015475\t0\n","ENSG00000025708\t0\n","ENSG00000025770\t0\n","ENSG00000040608\t0\n","ENSG00000054611\t0\n"]}]},{"cell_type":"markdown","source":["merge HTSeq-counts tables for further analysis with edgeR on R and Rstudio"],"metadata":{"id":"wd-REL75-m9R"}},{"cell_type":"code","source":["!join gene_counts_HTseq2.gff uhr2_gene_read_counts_table.tsv > gene_counts_HTseq3.gff"],"metadata":{"id":"LQnFzDR0oObO","executionInfo":{"status":"ok","timestamp":1673537999249,"user_tz":-120,"elapsed":265,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":82,"outputs":[]},{"cell_type":"code","source":["!join brain1_gene_read_counts_table.tsv brain2_gene_read_counts_table.tsv > gene_counts_HTseq.gff"],"metadata":{"id":"SE4wAGeW-Yku","executionInfo":{"status":"ok","timestamp":1673537851185,"user_tz":-120,"elapsed":238,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":77,"outputs":[]},{"cell_type":"markdown","source":["gene_counts_HTseq.gff has 5 extra lines which are the basic statistics.\n","\n","They should be removed for use as an input file for edgeR.\n","\n","They can be removed using the following command:"],"metadata":{"id":"x16PWTYq_E6d"}},{"cell_type":"code","source":["!sed '/^_/d' gene_counts_HTseq3.gff > gene_counts_HTseq.tab"],"metadata":{"id":"Pky-7cwF_Hhy","executionInfo":{"status":"ok","timestamp":1673538048890,"user_tz":-120,"elapsed":272,"user":{"displayName":"Mostafa Zaki","userId":"08817150632828476774"}}},"execution_count":83,"outputs":[]},{"cell_type":"markdown","source":["\n","\n","---\n","\n"],"metadata":{"id":"7z5h_e0hHYP3"}},{"cell_type":"markdown","source":["# FPKM-based Gene Expression Using Cufflinks Package\n","\n","To perform FPKM based gene expression, we use Cufflinks with the\n","following command:"],"metadata":{"id":"5gkTc2WcHeT6"}},{"cell_type":"code","source":["!cufflinks -p 2 -o brain1 --library-type fr-firststrand --GTF gtf/chr22.gtf --frag-lenmean \\\n","262 --frag-len-std-dev 80 --no-update-check star_SAM/brain1.bam\n","\n","!cufflinks -p 2 -o brain2 --library-type fr-firststrand --GTF gtf/chr22.gtf --frag-lenmean \\\n","262 --frag-len-std-dev 80 --no-update-check star_SAM/brain2.bam\n","\n","!cufflinks -p 2 -o uhr1 --library-type fr-firststrand --GTF gtf/chr22.gtf --frag-len-mean \\\n","262 --frag-len-std-dev 80 --no-update-check star_SAM/uhr1.bam\n","\n","!cufflinks -p 2 -o uhr2 --library-type fr-firststrand --GTF gtf/chr22.gtf --frag-len-mean \\\n","262 --frag-len-std-dev 80 --no-update-check star_SAM/uhr2.bam"],"metadata":{"id":"ErYnEG53HmwD"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["#Merge Cufflinks Assembly Using Cuffmerge\n","We will need to join the transcript assembly feature files generated\n","by Cufflinks to create a single assembly feature file. To do this, use\n","the following commands:"],"metadata":{"id":"jTcGXHfoImij"}},{"cell_type":"code","source":["!ls */transcripts.gtf > assembly_GTF_list.txt\n","!cuffmerge -p 2 -o merged -g gtf/chr22.gtf -s genome/chr22.fa assembly_GTF_list.txt"],"metadata":{"id":"6-BRe0N2IpOz"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["#Quantify Expression Using Cuffquant\n","Expressions are then quantified using Cuffquant with the following\n","commands:"],"metadata":{"id":"nNEAHvMiI3vp"}},{"cell_type":"code","source":["!cuffquant -p 2 --library-type fr-firststrand --frag-len-mean 262 --frag-len-std-dev \\\n","80 --no-update-check -o brain1 merged/merged.gtf star_SAM/brain1.bam\n","\n","!cuffquant -p 2 --library-type fr-firststrand --frag-len-mean 262 --frag-len-std-dev \\\n","80 --no-update-check -o brain2 merged/merged.gtf star_SAM/brain2.bam\n","\n","!cuffquant -p 2 --library-type fr-firststrand --frag-len-mean 262 --frag-len-std-dev \\\n","80 --no-update-check -o uhr1 merged/merged.gtf star_SAM/uhr1.bam\n","\n","!cuffquant -p 2 --library-type fr-firststrand --frag-len-mean 262 --frag-len-std-dev \\\n","80 --no-update-check -o uhr2 merged/merged.gtf star_SAM/uhr2.bam"],"metadata":{"id":"FBKsGn2JI6a7"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["#Differential Expression Using Cuffdiff"],"metadata":{"id":"bTtSKPzzJpLc"}},{"cell_type":"code","source":["!cuffdiff -p 2 -L brain,uhr -o ref_only/ --library-type fr-firststrand --no-update-check merged/merged.gtf \\\n","brain1/abundances.cxb,brain2/abundances.cxb uhr1/abundances.cxb,uhr2/abundances.cxb"],"metadata":{"id":"AlNO3q2kJur2"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["You can view the gene expression output on command line\n","using the following command:"],"metadata":{"id":"8ApzKChIKBQZ"}},{"cell_type":"code","source":["!less -S ref_only/gene_exp.diff"],"metadata":{"id":"Z4GDCiIOKBpn"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["To count the number of genes differential expressed via command\n","line, you can use the following command:"],"metadata":{"id":"-oS3C-ouKMLN"}},{"cell_type":"code","source":["!awk '{if($7 == \"OK\" && $10 <= 0.05) print $0}' ref_only/gene_exp.diff | wc -l\n","#To view the number of genes differential expressed via command line, you can use the following command:\n","!awk '{if($7 == \"OK\" && $10 <= 0.05) print $0}' ref_only/gene_exp.diff | less\n","#To extract significant differentially expressed gene(s):\n","!awk '{if($14 == \"yes\") print $0}' ref_only/gene_exp.diff > DE_genes.txt"],"metadata":{"id":"uVRzreklKV89"},"execution_count":null,"outputs":[]}]}